Modules/Tests/YosemiteTests/Tools/POS/POSRefundsServiceTests.swift

```diff
@@ -0,0 +1,173 @@
+import Testing
+import Foundation
+@testable import Yosemite
+import Networking
+
+struct POSRefundsServiceTests {
+    @Test func providePointOfSaleRefunds_then_calls_remote_with_expected_params() async throws {
+        // Given
+        let remote = MockPOSRefundsRemote()
+        let siteID: Int64 = 123
+        let sut = POSRefundsService(siteID: siteID, refundsRemote: remote)
+        let orderRefunds = [POSOrderRefundCondensed(refundID: 10, formattedTotal: "$22"), POSOrderRefundCondensed(refundID: 20, formattedTotal: "$22")]
+
+        let order = POSOrder(id: 1, number: "1001", dateCreated: Date(), status: .completed,
+                             formattedTotal: "$10.00", formattedSubtotal: "$10.00", customerEmail: "test1@example.com",
+                             paymentMethodTitle: "Credit Card", lineItems: [],
+                             refunds: orderRefunds, formattedDiscountTotal: nil, formattedTotalTax: "$0.00",
+                             formattedPaymentTotal: "$10.00", formattedNetAmount: nil)
+
+        // When
+        _ = try await sut.providePointOfSaleRefunds(for: order)
+
+        // Then
+        #expect(remote.spySiteID == siteID)
+        #expect(remote.spyOrderID == order.id)
+        #expect(remote.spyRefundIDs == orderRefunds.map { $0.refundID })
+    }
+
+    @Test func providePointOfSaleRefunds_when_remote_fails_then_propagates_remote_error() async throws {
+        // Given
+        let remote = MockPOSRefundsRemote()
+        let sut = POSRefundsService(siteID: 123, refundsRemote: remote)
+
+        struct TestError: Error {}
+        remote.result = .failure(TestError())
+
+        let order = POSOrder(id: 1, number: "1001", dateCreated: Date(), status: .completed,
+                             formattedTotal: "$10.00", formattedSubtotal: "$10.00", customerEmail: "test1@example.com",
+                             paymentMethodTitle: "Credit Card", lineItems: [],
+                             refunds: [], formattedDiscountTotal: nil, formattedTotalTax: "$0.00",
+                             formattedPaymentTotal: "$10.00", formattedNetAmount: nil)
+
+        // Then
+        do {
+            _ = try await sut.providePointOfSaleRefunds(for: order)
+            Issue.record("Expected error to be thrown")
+        } catch {
+            #expect(error is TestError)
+        }
+    }
+
+    @Test func providePointOfSaleRefunds_when_remote_succeeds_then_returns_same_count_as_remote() async throws {
```

Comment:
I would expect this test to fail in the remote since we're calling `MockRefunds.sampleRefund()` twice, hence they have the same refund ID = 0. AFAU we should handle both cases at this layer or in the networking one (remote failure and remote success)